File: #1-25 (C, Java, Python)/Java/Euler24.java
//Euler24 ~ Lexicographic permutations
import java.util.ArrayList;
import java.util.Collections;
import java.util.concurrent.TimeUnit;

public class Euler24
{
	static ArrayList<Long> permutations = new ArrayList<Long>();

	public static void main(String args[])
	{
		long startTime = System.nanoTime();
		
		permutation("","0123456789");
		Collections.sort(permutations);
		long solution = permutations.get(1000000-1);

		long endTime = System.nanoTime();
		long timeElapsed = endTime - startTime;
		System.out.println("Solution: " + solution + "\n");
		System.out.println("Program completed in " + timeElapsed/1000000000 + " seconds.");		
	}

	//Use recursion to count all permutations
	public static void permutation(String prefix, String str)
	{
		int n = str.length();
		if (n == 0) 
		{
			permutations.add(Long.parseLong(prefix));
		}
		else
		{
			for (int i = 0; i < n; i++)
			{
				permutation(prefix + str.charAt(i), str.substring(0, i) + str.substring(i+1, n));
			}
		}
	}
}

File: #1-25 (C, Java, Python)/Python/Euler24.py
#Euler24 ~ Lexicographic permutations
import timeit
from math import factorial

#Use recursion to count all permutations
def perm(n, s):
    if(len(s) == 1):
        return s
    
    q, r = divmod(n, factorial(len(s)-1))
    return s[q] + perm(r, s[:q] + s[q+1:])

start = timeit.default_timer()

solution = perm(999999, '0123456789') #Get 1 millionth permutation

stop = timeit.default_timer()
print("Solution: ",solution,"\n")
print("Program completed in","{0:.3f}".format(stop - start), "seconds.")